lib/sinon/spy.js

```diff
@@ -271,10 +276,11 @@ var spyApi = {
         var args = slice.call(arguments);
 
         if (this.fakes) {
-            var match = matchingFake(this.fakes, args, true);
+            var matchings = this.matchingFakes(args, true);
+            var matching = matchings && matchings.pop();
```

Comment:
If `matchingFakes` always returns arrays, you don't need to test for the result first, and can just use `matchings.pop()` directly.